### PR TITLE
Replace unmaintained parity-secp256k1 to secp256k1

### PR DESCRIPTION
**Pull request by ahlinc**
This PR replaces an unmaintained fork of secp256k1 to the latest version of the original library that already have all required features with a more clear API. Also this will solve compilation issues with llvm 13.* tools on MacOS like the next: https://github.com/rust-lang/rust-bindgen/issues/1834

### CODE REVIEW

File: core/crypto/src/signature.rs [hunk 1]
```diff
@@ -940,18 +936,18 @@ mod tests {
 
         let sk = SecretKey::from_seed(KeyType::SECP256K1, "test");
         let pk = sk.public_key();
-        let expected = "\"secp256k1:BtJtBjukUQbcipnS78adSwUKE38sdHnk7pTNZH7miGXfodzUunaAcvY43y37nm7AKbcTQycvdgUzFNWsd7dgPZZ\"";
+        let expected = "\"secp256k1:2xVqteU8PWhadHTv99TGh3bSffGb32cFokopSg8xPAiFVjv5UvRUYfNPzmZctBuXryhQosc9dfY1TKKrV4YiyiZH\"";
```
Comment (mina86) [inline core/crypto/src/signature.rs:939]: This looks a bit sus.  Do we depend anywhere in the protocol on how `from_seed` generates keys?

**Comment by longnguyencircle** [comment]
I'm getting this error.

```
cargo build --target wasm32-unknown-unknown --release                                                             101 ↵
   Compiling parity-secp256k1 v0.7.0
   Compiling toml v0.5.9
   Compiling wasm-bindgen-macro v0.2.82
   Compiling serde_json v1.0.83
   Compiling near-rpc-error-core v0.14.0
   Compiling num-rational v0.3.2
   Compiling curve25519-dalek v3.2.1
The following warnings were emitted during compilation:

warning: depend/secp256k1/contrib/lax_der_parsing.c:7:10: fatal error: 'string.h' file not found
warning: #include <string.h>
warning:          ^~~~~~~~~~
warning: 1 error generated.

error: failed to run custom build command for `parity-secp256k1 v0.7.0`
```

**Comment by mina86** [comment]
This doesn’t look related to parity-secp256k1.  It appears you’re missing wasm32 libc header files.  Why / what are you compiling for wasm32 anyway?

File: core/crypto/src/signature.rs (hunk 1 above)
Comment (mina86) [reply core/crypto/src/signature.rs]: @mm-near, do you know of any issues we might face if we switch the RNGs in secp256k1 package? From what I’m seeing this is only affecting tests and the PR LGTM.

File: core/crypto/src/signature.rs [hunk 2]
```diff
@@ -940,18 +936,18 @@ mod tests {
 
         let sk = SecretKey::from_seed(KeyType::SECP256K1, "test");
         let pk = sk.public_key();
-        let expected = "\"secp256k1:BtJtBjukUQbcipnS78adSwUKE38sdHnk7pTNZH7miGXfodzUunaAcvY43y37nm7AKbcTQycvdgUzFNWsd7dgPZZ\"";
+        let expected = "\"secp256k1:5ftgm7wYK5gtVqq1kxMGy7gSudkrfYCbpsjL6sH1nwx2oj5NR2JktohjzB6fbEhhRERQpiwJcpwnQjxtoX3GS3cQ\"";
```
Comment (mm-near) [inline core/crypto/src/signature.rs:939]: FYI - this might also impact our localnet runs -- as that's the place where we generate keys from seed.

**Review by mm-near — APPROVED** [review]

**Review by mina86 — APPROVED** [review]

**Pull request merged**